common/v2/services/EthService/__tests__/validators.spec.ts

```diff
@@ -25,3 +26,68 @@ describe('isTransactionFeeHigh', () => {
     expect(isTransactionFeeHigh('0.001', 300, false, '21000*', '20a')).toBe(false);
   });
 });
+
+describe('isValidETHRecipientAddress', () => {
+  it('returns true for a valid checksummed address', () => {
+    const expected = { success: true };
+    const testAddress = '0x4bbeEB066eD09B7AEd07bF39EEe0460DFa261520';
+    const returned = isValidETHRecipientAddress(testAddress, undefined);
+    expect(returned.success).toBe(expected.success);
+  });
+  it('returns true for a valid all-lowercase address', () => {
+    const expected = { success: true };
+    const testAddress = '0x4bbeeb066ed09b7aed07bf39eee0460dfa261520';
+    const returned = isValidETHRecipientAddress(testAddress, undefined);
+    expect(returned.success).toBe(expected.success);
+  });
+  it('returns true for a valid all-uppercase address', () => {
+    const expected = { success: true };
+    const testAddress = '0X4BBEEB066ED09B7AED07BF39EEE0460DFA261520';
+    const returned = isValidETHRecipientAddress(testAddress, undefined);
+    expect(returned.success).toBe(expected.success);
+  });
+  it('returns true for a valid ens name', () => {
+    const expected = { success: true };
+    const testAddress = 'mycryptoid.eth';
+    const returned = isValidETHRecipientAddress(testAddress, undefined);
+    expect(returned.success).toBe(expected.success);
+  });
+  it('returns false for an invalid ens name', () => {
+    const expected = { success: false };
+    const testAddress = 'mycryptoid.ethhhhh';
+    const returned = isValidETHRecipientAddress(
+      testAddress,
+      ('Domain mycryptoid.ethhhh is not supported' as unknown) as ResolutionError
+    );
+    expect(returned.success).toBe(expected.success);
+  });
+  it('returns false for an unresolved ens name', () => {
```

Comment:
What stops someone from "grief attacking" us by registering this ENS name to slow down a release - maybe even if its a hotfix release? or am I too paranoid?

Also the `testAddress` and the `returned` string have different ENS names in it - not sure if that's a _problem_.

Reply:
woopsies